<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!--Start Gazebo-->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <!--Spawn Robot-->
    <arg name="model" default="$(find balance_robot_description)/urdf/balance_robot.xacro" />
    <param name="robot_description" command="$(find xacro)/xacro $(arg model) --inorder" />
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model balance_robot"/>

    <!--Start Robot Controller -> diff drive controller -->
    <rosparam command="load" file="$(find balance_robot_description)/config/diff_drive_controller.yaml"  ns="/balance_robot"/>
    <node name="controller_spawner" ns="/balance_robot" output="screen" pkg="controller_manager" 
        respawn="false" type="spawner" args="diff_drive_controller"/>
    <!--Not required in diff_drive_controller 
        <rosparam file="$(find balance_robot_description)/config/gazebo_ros_control.yaml" command="load" />-->

    <!--Publish joint state(gazebo info), robot state(tf) -->
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    <!--Start rviz-->
    <arg default="$(find balance_robot_description)/launch/urdf.rviz" name="rvizconfig"/>
    <node args="-d $(arg rvizconfig)" name="rviz" pkg="rviz" required="true" type="rviz"/>

</launch>